test/src/test/kotlin/graphql/nadel/tests/EngineTests.kt

```diff
@@ -165,25 +167,35 @@ private suspend fun execute(
                     return ServiceExecution { params ->
                         try {
                             val incomingQuery = params.query
-                            val incomingQueryPrinted = AstPrinter.printAst(
+                            val actualVariables = params.variables
+                            val actualOperationName = params.operationDefinition.name
+                            val actualQuery = AstPrinter.printAst(
                                 astSorter.sort(incomingQuery),
                             )
-                            printSyncLine(incomingQueryPrinted)
+                            printSyncLine(actualQuery)
 
                             val response = synchronized(serviceCalls) {
                                 val indexOfCall = serviceCalls
                                     .indexOfFirst {
                                         it.serviceName == serviceName
-                                            && AstPrinter.printAst(it.request.document) == incomingQueryPrinted
-                                            && it.request.operationName == params.operationDefinition.name
-                                            && it.request.variables == params.variables
+                                            && AstPrinter.printAst(it.request.document) == actualQuery
+                                            && it.request.operationName == actualOperationName
+                                            && it.request.variables == actualVariables
                                     }
                                     .takeIf { it != -1 }
 
                                 if (indexOfCall != null) {
                                     serviceCalls.removeAt(indexOfCall).response
                                 } else {
-                                    fail("Unable to match service call")
+                                    fail(
+                                        """Unable to match service call 
+                                        |   fixture : '${fixture.name}' 
+                                        |   service : '${serviceName}' 
+                                        |   query : '${actualQuery}' 
+                                        |   variables : '${actualVariables}' 
+                                        |   operation : '${actualOperationName}' 
+                                        """.trimMargin()
+                                    )
```

Comment:
This makes the fail message a smidge easier to work out whats gone wrong.  I know its printed earlier but its now all near the last exception

Reply:
Nice!